x-pack/plugins/observability/public/hooks/use_query_params.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useLocation } from 'react-router-dom';
+import { useMemo } from 'react';
+import { parse } from 'query-string';
+import { UI_SETTINGS, useKibanaUISettings } from './use_kibana_ui_settings';
+import { TimePickerTime } from '../components/shared/data_picker';
+import { getAbsoluteTime } from '../utils/date';
+
+const getParsedParams = (search: string) => {
+  return search ? parse(search[0] === '?' ? search.slice(1) : search, { sort: false }) : {};
+};
+
+export function useQueryParams() {
```

Comment:
I honestly don't see a reason why to have such a hook, I believe the current implementation is sufficient for now, where each page receives its own params and pass it down one level to the data section components. If one day we feel that we have to pass params more down in the tree than I would be ok with that. WDYT?

Reply:
I got what you meant by doing it... it's because of `absoluteTime` and `relativeTime` are objects, that's why it rerenders the APM/Logs/Uptime... components.